python/ray/serve/tests/test_deploy.py

```diff
@@ -57,6 +57,42 @@ def d():
     assert pid3 != pid2
 
 
+@pytest.mark.parametrize(
+    "component_name",
+    [
+        "test@component",
+        "test#123",
+        "component/name",
+        "component.name",
+        "component!name",
+        "component$name",
+        "component%name",
+        "component^name",
+        "component&name",
+        "component*name",
+    ],
+)
+def test_deploy_with_special_characters(serve_instance, component_name):
+    """The function should not fail when the deployment name contains special characters."""
+    signal = SignalActor.options(name=f"signal-{get_random_string()}").remote()
+
+    # V1 blocks on signal
+    @serve.deployment(name="some_name")
+    class V1:
+        async def handler(self):
+            await signal.wait.remote()
+            return 1, os.getpid()
+
+        async def __call__(self):
+            return await self.handler()
+
+    # Check that deployment succeeds with special characters
+    with pytest.raises(Exception) as exc_info:
+        serve.run(V1.options(name=component_name).bind(), name="app")
```

Comment:
can you add a similar test for application_name because that can also have similar problem.